webapp/components/self-navigator.html

```diff
@@ -11,17 +11,19 @@
             value: '/',
             observer: 'pathUpdated',
           },
+          encodedPath: {
+            type: String,
+            computed: 'computeEncodedPath(path)'
+          },
           onpathupdated: Function,
+          encodeTestPath: Function,
+          decodeTestPath: Function,
```

Comment:
I don't think these are needed; just always encode/decode the value as separated properties (`encodedPath` and `path`).

Reply:
Done.